### PR TITLE
Encrypted Volumes

### PR DESCRIPTION
**Pull request by dwlehman**
This adds support for LUKS volumes. ~~The testing is very minimal. I'm curious what it will take to get it to pass CI.~~

### CODE REVIEW

**Comment by dwlehman** [comment]
This is built on top of my swap-minimal branch (from #90). Only the last five commits are specific to this.

**Comment by richm** [comment]
@sergio-correia FYI

**Comment by dwlehman** [comment]
Weird that the luks tests should be what fails since that's what I ran locally on F32 prior to opening/updating this pull request.

**Comment by dwlehman** [comment]
Oh, good grief. It only passes on the OS I developed and initially tested on? It'll have to wait til next week.

**Comment by dwlehman** [comment]
What's missing at this point is unknown, aside from better testing:

1. verify that the LUKS layer exists as appropriate and has the expected name
2. verify the LUKS parameters insofar as that is possible

File: README.md [hunk 1]
```diff
@@ -73,6 +73,27 @@ The `mount_point` specifies the directory on which the file system will be mount
...
+This string specifies a passphrase used to unlock/open the LUKS volume.
+
+##### `encryption_key_file`
+This string specifies the full path to the key file used to unlock the LUKS volume.
+
+##### `encryption_cipher`
+This string specifies a non-default cipher to be used by LUKS.
+
+##### `encryption_key_size`
+This integer specifies the LUKS key size (in bytes).
```
Comment (vojtechtrefny) [inline README.md:92]: The key size should be in bits, e.g. 512

Comment (dwlehman) [reply README.md]: Doh, good catch -- thanks!

**Pull request merged**